Update the nix shell paragraph in the README (#16)
haskellfoundation/matchmaker: README.md

* Yarn 1.22 or higher

*Note*
There is a `shell.nix` file provided for convenience. However it is far from perfect.
It will not manage a local installation of PostgreSQL for you.
You should be able to work in a pure shell. If not, file a bug!

> The nix shell should source all environment variables when you enter it. Additionally it provides 3 conveniences scripts:

Reviewer: ```suggestion
The nix shell should source all environment variables when you enter it. Additionally it provides 3 convenience scripts:
```

Reviewer: whoops typo on my part!

* Yarn 1.22 or higher

*Note*
There is a `shell.nix` file provided for convenience. However it is far from perfect.

> It will ont manages a local installation of PostgreSQL for you (yet).

Reviewer: ```suggestion
It will not manage a local installation of PostgreSQL for you.
```

Reviewer: Nix shell actually can't install postgres for you unless you want to start with a fresh postgres everytime you enter nix-shell (which is generally infeasible). We can direct people to this config though:

```nix
  services.postgresql = {
    package = pkgs.postgresql_12;
    enable = true;
    enableTCPIP = false;
    authentication = ''
      local all all trust
      host all all 127.0.0.1/32 trust
      host all all ::1/128 trust
    '';
    settings = {
      timezone = "UTC";
      shared_buffers = 128;
      fsync = false;
      synchronous_commit = false;
      full_page_writes = false;
    };
  };
```

Reviewer: this is only for NixOS (maybe nix-darwin too)

Author: Yeah, IHP handles the postgres startup from the nix env too. I just wanted to clarify the limits of the nix shell. :)

You should be able to work in a pure shell. If not, file a bug!

> 

Reviewer: ```suggestion
The nix shell should source all environment variables when you enter it. Additionally it provides 3 conveniences scripts:

1.  `dev` - This will run ghcid with a couple flags enabled
2.  `run` - This will run the matchmaker application
3.  `format` - This will run `stylish-haskell` on all .hs files in the current directory and below

```
